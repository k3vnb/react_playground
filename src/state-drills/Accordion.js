import React from 'react'
import './Accordion.css'


class Accordion extends React.Component {
    state = {
        isShown: null
    }



    handleClick = i => {
        this.state.isShown === i ? 
        this.setState({
            isShown: null
        }) :
        this.setState({
            isShown: i
        })
    }

    render() {
        const liList = this.props.sections.map((item, i) => <li key={i}><button id={`${i}`} onClick={() => this.handleClick(i)}>{item.name}</button><p className={this.state.isShown === i ? 'show-content' : 'hide-content'}>{item.content}</p></li>)
        return (
            <div>
                <hr/>
                <h2>Accordion</h2>
                <ul>
                    {!!this.props.sections.length && liList}
                </ul>
            </div>
        );
    }

    static defaultProps = {
        sections: []
    }
}

export default Accordion;