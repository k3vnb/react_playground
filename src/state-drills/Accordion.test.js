import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe('Accordion component', () => {

    const accordionProps = [
        { name: 'First tab',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.' },
        { name: 'Second tab',
          content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.' },
        { name: 'Third tab',
          content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.' },
      ];

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Accordion />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders empty given no selection', () => {
        const wrapper = shallow(<Accordion sections={accordionProps}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the first tab and opens any clicked tab', () => {
      const wrapper = shallow(<Accordion sections={accordionProps} />)
      wrapper.find('button')
      wrapper.find('button').at(1).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('has no tabs open when the current open tab has its button clicked', () => {
      const wrapper = shallow(<Accordion sections={accordionProps} />)
      wrapper.find('button')
      wrapper.find('button').at(0).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
    })
})