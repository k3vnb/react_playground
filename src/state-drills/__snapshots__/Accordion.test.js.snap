// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component closes the first tab and opens any clicked tab 1`] = `
<div>
  <hr />
  <h2>
    Accordion
  </h2>
  <ul>
    <li
      key="0"
    >
      <button
        id="0"
        onClick={[Function]}
      >
        First tab
      </button>
      <p
        className="hide-content"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        id="1"
        onClick={[Function]}
      >
        Second tab
      </button>
      <p
        className="show-content"
      >
        Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="2"
    >
      <button
        id="2"
        onClick={[Function]}
      >
        Third tab
      </button>
      <p
        className="hide-content"
      >
        Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion component has no tabs open when the current open tab has its button clicked 1`] = `
<div>
  <hr />
  <h2>
    Accordion
  </h2>
  <ul>
    <li
      key="0"
    >
      <button
        id="0"
        onClick={[Function]}
      >
        First tab
      </button>
      <p
        className="show-content"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        id="1"
        onClick={[Function]}
      >
        Second tab
      </button>
      <p
        className="hide-content"
      >
        Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="2"
    >
      <button
        id="2"
        onClick={[Function]}
      >
        Third tab
      </button>
      <p
        className="hide-content"
      >
        Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion component renders empty given no selection 1`] = `
<div>
  <hr />
  <h2>
    Accordion
  </h2>
  <ul />
</div>
`;
